### ISSUE TITLE
Images Are Not Optimized In dest()

### ISSUE THREAD

**Issue by CelticParser** [imagemin/imagemin#138]
I just tried out `imagemin` and I am finding the output files are not minimized/optimized. The files are being written to `.dist()` but the .jpg's and .png's are only being duplicated from the `.src()`. Also, .svg's are copied w/o data. 

I may just be doing something wrong(?).

**Usage:**

    new Imagemin()
    .src(environment + '/content/img/*.{gif,jpg,png,svg}')
    .dest(environment + '/_public/img/')
    .use(Imagemin.jpegtran({
      progressive: true
    }))
    .use(Imagemin.optipng({
      optimizationLevel: 5
    }))
    .use(Imagemin.svgo())
    .run(function (err, files) {
      if (err) {
        console.log(err);
      } else {
        console.log(files);
      }
    });

I have tried changing the `optimizationLevel` for .png's and played with the `progressive` properties for the .jpg's and no change.  If I run the images thru an online processor, I get a min of 30% reductions.

**Log:**

```
{ [Error: spawn /home/ubuntu/workspace/_op/node_modules/imagemin/node_modules/imagemin-jpegtran/node_modules/jpegtran-bin/vendor/jpegtran ENOENT]
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'spawn /home/ubuntu/workspace/_op/node_modules/imagemin/node_modules/imagemin-jpegtran/node_modules/jpegtran-bin/vendor/jpegtran',
  path: '/home/ubuntu/workspace/_op/node_modules/imagemin/node_modules/imagemin-jpegtran/node_modules/jpegtran-bin/vendor/jpegtran',
  spawnargs: [ '-copy', 'none', '-optimize', '-progressive' ],
  fileName: '/home/ubuntu/workspace/env/production/content/img/Jason-Robert-OKennedy.jpg' }
[Error: no writecb in Transform class]
```

Help  will be greatly appreciated.

**Comment by shinnn** [comment]
Can you create a small repository by which we can reproduce the problem? That is most helpful for us to find the root cause.

**Comment by CelticParser** [comment]
Sure can.  Give me a few.  Are you on [C9](c9.io)?  If not, I can give access to a cloud IDE dev environment  using your GitHub account email (@shinnn).

**Comment by shinnn** [comment]
Why not just a public repo on Github?

**Comment by CelticParser** [comment]
I'll send you both.  Just always best to duplicate exact environments for debugging.

I did find out to why the files were being copied to the _public directory: One of my build scripts was copying everything and that is fixed now. However the same error is being thrown.

Repo: https://github.com/CelticParser/imagemin-example.git
C9:  https://ide.c9.io/jason_alan_kennedy/imagemin

FYI: We can collaborate (live) on c9   

Development Environment:
Node:     v4.2.1
npm:      v2.14.7
Ubuntu: 4.04.3 LTS

**Comment by shinnn** [comment]
No binaries are installed. Usually `vendor/jpegtran` is created [here](https://github.com/CelticParser/imagemin-example/tree/884f2630855d5434c7dbd5eeabe413daafe111a6/_op/node_modules/imagemin/node_modules/imagemin-jpegtran/node_modules/jpegtran-bin).

Can you remove `node_module` folder, re-install imagemin and show us the entire log?

**Comment by shinnn** [comment]
Also, does your network is behind proxy and is any environment variable for proxy setting (e.g. `HTTPS_PROXY`) enabled?

**Comment by CelticParser** [comment]
I discovered the cause of issue. Within your `package.json` I discovered the following is declared:
```
"engines": {
    "node": ">=0.10.0"
  }
```

I see a lot of DevOps doing this.

As stated in [this comment](https://github.com/imagemin/imagemin/issues/138#issuecomment-159379495), we are using Node v4.2.1 for security reason (_There's some serious one's after v0.10.x_).

I uninstalled `imagemin` with NPM and cleared npm's cache. I next used nvm to revert back to **v0.10.40 (defaults to npm v1.4.28)** and reinstalled `imagemin`. The module works as expected :). I repeated the procedure and continued up the version chain and when I installed with **v0.12.7 (defaults to npm v2.11.3)**, this module breaks :8ball:  I went back to v0.10.40 and it works.

The package should be changed to:

 ```
"engines": {
    "node": "^0.10.x"
  }
```

I am thinking that it is most-likely the npm version than node's.  At version node 0.12.7, the test script does not run. I don't have time to test witch npm version the module breaks until after the weekend.  If you have Ideas where in the code it could be (it may be a dependency - not good), it could be an easy fix but until then it might be wise to set the package to:

```
"engines": {
    "node": "^0.10.x",
    "npm": ^2.1.5"
  }
```

@shinnn you still have access to the IDE - It will make Node and npm switchin' like eating cake.

Thnx

**Comment by CelticParser** [comment]
**P.S.** Having `"node": ">=0.10.0"` does not throw warnings by the npm-cli if the user is running above "node": ">=0.10.0", hence the `>=`.

**Comment by shinnn** [comment]
Weird. [This project passes the test with Node v4.2.1 and npm v2.14.7](https://travis-ci.org/imagemin/imagemin/jobs/86530458) (and [other combinations](https://travis-ci.org/imagemin/imagemin/builds/88269138)). Also I'm using imagemin with Node v5.1.0 and npm v3.3.12 and it works fine.

No need to change `engines` field. This seems like an issue with your development environment.

**Comment by CelticParser** [comment]
Interesting. I'll  run it with Node v5.1.0 and npm v3.3.12 and if it fails,
I'll check with c9 support and see if they have any Ideas. Thnx

**Comment by shinnn** [comment]
Is there any error logs on `npm install imagemin`?

**Comment by CelticParser** [comment]
No, there is no error logged. The only error thrown is the one I added to your script that I posted above.

My question is are you only relying on what the Travis build returns? Second, are you or your QA dept. verifying that the generated files thru the `dest()` function is being compressed?

There is something weird going on here because we tested this with numerous environments today and on different machines and in cloud environments and are seeing the same thing: It works with some Node/npm combos and others it doesn't.  We verified that when it doesn't, it doesnt on any system/workstation and our cloud dev environment.  

When you showed that the Travis build is passing, that raised a flag. I forked the project and ran the test and ran it locally and on a private Travis server. It passed with all setup combinations in the travis.yml.  However, when I physically check the generated file in the destination directory, it was not compressed - including running with Node v5.1.0 and npm v3.3.12.

I  pulled a snippet from your test.js and reached out to the community to see if there is an error or known issue with the format being used or a better way to write the unit-test.  I didn't get any replies so I reached out directly to the folks at Travis.  I got a reply that the test "does not correctly" compare file sizes and will pass. I could not get a reply (as of yet) on to why or how to correct it.

So we decided to fork the project and cloned it into our private repo for further development for our use under the MIT license.  We plan on eliminating all the co-dependencies (75!) for better maintainability and security. If we find what is the cause of the issue, someone on our team will release the patch via a pull request.

Thnx
